contracts/src/factory/facets/architect/IArchitect.sol

```diff
@@ -42,6 +42,24 @@ interface IArchitectBase {
     string longDescription;
   }
 
+  struct Metadata {
+    string name;
+    string uri;
+    string shortDescription;
+    string longDescription;
+  }
+
+  struct Prepay {
+    uint256 supply;
+  }
+
+  struct CreateSpace {
+    Metadata metadata;
+    Membership membership;
+    ChannelInfo channel;
+    Prepay prepay;
+  }
```

Comment:
```suggestion
  struct CreateSpace {
    SpaceInfo spaceInfo;
    Prepay prepay;
  }
```
Or consider this to use calldata without reconstructing `SpaceInfo` in memory.

Reply:
It doesn't change `SpaceInfo`. By keeping it nested in `CreateSpace` we avoid reconstructing `SpaceInfo` in memory.

Reply:
you have a nested spaceInfo key when creating the client object:

```
{
  "spaceInfo": {},
  "prepay": {}
}
```

instead of

```
{
  metadata: "",
  membership: "",
  channel: "",
  prepay: ""
}
```